core/sql/optimizer/NormRelExpr.cpp

```diff
@@ -2404,8 +2412,13 @@ void Join::createAFilterGrandChildIfNeeded(NormWA & normWARef)
   if (doNotUnnest == FALSE)
   {
     nonLocalPreds.clear();
-    if (oldRightGrandChild->selectionPred().getReferencedPredicates
-                                        (outerReferences, nonLocalPreds))
+    oldRightGrandChild->selectionPred().getReferencedPredicates
+      (outerReferences, nonLocalPreds) ;
+    if (nestedAggInSubQ)
+      subQGby->selectionPred().getReferencedPredicates
+	(outerReferences, nonLocalPreds);
+
+    if (!nonLocalPreds.isEmpty())
```

Comment:
I think for a groupby by the time we get past the binder the having clause is the same as the where clause. In other words the predicate is kept in selectionPred() of RelExpr base class. It is applied at the end of the work method before a result row is sent to parent. From a SQL syntax point of view this works for HAVING, since now the selectionPred() can reference aggregates evaluated in this GB.